service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/PractitionerAgentPersonMapperTest.java

```diff
@@ -0,0 +1,107 @@
+        messageContext = new MessageContext(randomIdGeneratorService);
+        OrganizationToAgentMapper organizationToAgentMapper = new OrganizationToAgentMapper(messageContext);
+        practitionerAgentPersonMapper = new PractitionerAgentPersonMapper(messageContext, organizationToAgentMapper);
+        fhirParseService = new FhirParseService();
+    }
+
+    @ParameterizedTest
+    @MethodSource("readPractitionerRoleTests")
+    public void When_MappingPractitionerToAgent_Expect_RepresentedAgentXml(String practitionerRoleJson, String outputXml)
+        throws IOException {
+
+        var jsonInputPractitionerRole = ResourceTestFileUtils.getFileContent(practitionerRoleJson);
+        var expectedOutput = ResourceTestFileUtils.getFileContent(outputXml);
+
+        Practitioner practitioner = getPractitionerResource();
+        PractitionerRole practitionerRole = fhirParseService.parseResource(jsonInputPractitionerRole, PractitionerRole.class);
+        Organization organization = getOrganizationResource();
+
+        var outputMessage = practitionerAgentPersonMapper
+            .mapPractitionerToAgentPerson(practitioner, Optional.of(practitionerRole), Optional.of(organization));
+        assertThat(outputMessage).isEqualToIgnoringWhitespace(expectedOutput);
+    }
+
+    @ParameterizedTest
+    @MethodSource("readPractitionerOnlyTests")
+    public void When_MappingPractitionerOnlyToAgent_Expect_NoRepresentedOrganization(String inputJson, String outputXml)
+        throws IOException {
+        var jsonInputPractitioner = ResourceTestFileUtils.getFileContent(inputJson);
+        var expectedOutput = ResourceTestFileUtils.getFileContent(outputXml);
+
+        Practitioner practitioner = fhirParseService.parseResource(jsonInputPractitioner, Practitioner.class);
+        var outputMessage = practitionerAgentPersonMapper
+            .mapPractitionerToAgentPerson(practitioner, Optional.empty(), Optional.empty());
+        assertThat(outputMessage).isEqualToIgnoringWhitespace(expectedOutput);
+    }
+
+    private static Stream<Arguments> readPractitionerOnlyTests() {
+        return TestArgumentsLoaderUtil.readTestCases(PRACTITIONER_ONLY_FILE_LOCATION);
+    }
+
+    private static Stream<Arguments> readPractitionerRoleTests() {
+        return TestArgumentsLoaderUtil.readTestCases(PRACTITIONER_ROLE_FILE_LOCATION);
+    }
+
+    private static Practitioner getPractitionerResource() throws IOException {
+        String jsonPractitioner = ResourceTestFileUtils.getFileContent(PRACTITIONER);
+        return new FhirParseService().parseResource(jsonPractitioner, Practitioner.class);
+    }
+
+    private static Organization getOrganizationResource() throws IOException {
+        String jsonOrganization = ResourceTestFileUtils.getFileContent(ORGANIZATION);
+        return new FhirParseService().parseResource(jsonOrganization, Organization.class);
+    }
+
+    @AfterEach
+    public void tearDown() {
```

Comment:
Remember. Pubic methods always go before private ones. And static ones go before instance ones.